gammapy/datasets/map.py

```diff
@@ -1230,7 +1234,12 @@ def write(self, filename, overwrite=False):
 
     @classmethod
     def _read_lazy(cls, name, filename, cache, format=format):
-        kwargs = {"name": name}
+        ds_name = make_name(name)
+        hdulist = fits.open(str(make_path(filename)))
```

Comment:
As you open the hdulist here anyway there is not advantage in putting it in `_read_lazy`. So the extortion of the name could just always happen in `MapDataset.read()`

Reply:
Just remove the modification here...

Reply:
In the read function, the name management is made for the non-lazy method. In the case of the lazy one, why we should not manage it?